internal/services/firewall/firewall_resource.go

```diff
@@ -578,6 +616,33 @@ func resourceFirewallDelete(d *pluginsdk.ResourceData, meta interface{}) error {
 	return err
 }
 
+func expandAutoscaleConfiguration(d map[string]interface{}) (azurefirewalls.AzureFirewallAutoscaleConfiguration, error) {
```

Comment:
By convention, this function shall accept a `[]interface{}`, which corresponds to `d.Get("autoscale_configuration")`. Also, it returns a pointer to the underlying SDK type.